Content.Server/_DV/CosmicCult/RogueAscendedObjectiveSystem.cs

```diff
@@ -0,0 +1,30 @@
+using Content.Server._DV.CosmicCult.Components;
+using Content.Server.Objectives.Systems;
+using Content.Shared.Objectives.Components;
+
+namespace Content.Server._DV.CosmicCult;
+
+public sealed class RogueAscendedObjectiveSystem : EntitySystem
+{
+    [Dependency] private readonly NumberObjectiveSystem _number = default!;
+
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<RogueInfectionConditionComponent, ObjectiveGetProgressEvent>(OnGetInfectionProgress);
+    }
+
+    private void OnGetInfectionProgress(EntityUid uid, RogueInfectionConditionComponent comp, ref ObjectiveGetProgressEvent args)
+    {
+        args.Progress = InfectionProgress(comp, _number.GetTarget(uid));
+    }
+
+    private float InfectionProgress(RogueInfectionConditionComponent comp, int target)
```

Comment:
This function is only used once and doesn't reference any fields, just inline it into `OnGetInfectionProgress`